ui/app/components/calendar-widget.js

```diff
@@ -3,139 +3,115 @@ import layout from '../templates/components/calendar-widget';
- * />
+ * <CalendarWidget @startTimestamp={{this.startTime}} @endTimestamp={{this.endTime}} @selectMonth={{this.handleSelection}} />
  *
+ *    @param {string} startTimestamp - ISO timestamp string of the calendar widget's start time, displays in dropdown trigger
+ *    @param {string} endTimestamp - ISO timestamp string for the calendar widget's end time, displays in dropdown trigger
+ *    @param {function} selectMonth - callback function from parent - fires when selecting a month or clicking "Current billing period"
+ *  />
  * ```
  */
 class CalendarWidget extends Component {
   currentDate = new Date();
-  currentYear = this.currentDate.getFullYear(); // integer
-  currentMonth = parseInt(this.currentDate.getMonth()); // integer and zero index
-
-  @tracked allMonthsNodeList = [];
-  @tracked displayYear = this.currentYear; // init to currentYear and then changes as a user clicks on the chevrons
+  @tracked calendarDisplayDate = this.currentDate; // init to current date, updates when user clicks on calendar chevrons
   @tracked showCalendar = false;
   @tracked tooltipTarget = null;
   @tracked tooltipText = null;
 
-  get selectedMonthId() {
-    if (!this.args.endTimeFromResponse) return '';
-    const [year, monthIndex] = this.args.endTimeFromResponse;
-    return `${monthIndex}-${year}`;
+  // both date getters return a date object
+  get startDate() {
+    return parseAPITimestamp(this.args.startTimestamp);
+  }
+  get endDate() {
+    return parseAPITimestamp(this.args.endTimestamp);
+  }
+  get getDisplayYear() {
+    return this.calendarDisplayDate.getFullYear();
   }
   get disableFutureYear() {
-    return this.displayYear === this.currentYear;
+    return isSameYear(this.calendarDisplayDate, this.currentDate);
   }
   get disablePastYear() {
-    let startYear = parseInt(this.args.startTimeDisplay.split(' ')[1]);
-    return this.displayYear === startYear; // if on startYear then don't let them click back to the year prior
+    // calendar widget should only go as far back as the passed in start time
+    return isSameYear(this.calendarDisplayDate, this.startDate);
   }
   get widgetMonths() {
-    const displayYear = this.displayYear;
-    const currentYear = this.currentYear;
-    const currentMonthIdx = this.currentMonth;
-    const [startMonth, startYear] = this.args.startTimeDisplay.split(' ');
-    const startMonthIdx = this.args.arrayOfMonths.indexOf(startMonth);
-    return this.args.arrayOfMonths.map((month, idx) => {
-      const monthId = `${idx}-${displayYear}`;
+    const startYear = this.startDate.getFullYear();
+    const startMonthIdx = this.startDate.getMonth();
+    return ARRAY_OF_MONTHS.map((month, index) => {
       let readonly = false;
 
       // if widget is showing billing start year, disable if month is before start month
```

Comment:
Since the component is no longer specific to client counts maybe this comment can be tweaked?